### PR TITLE
[BugFix] Fix error of storage volume name in grants table

### LINKED ISSUE #26856: [storage volume] storage volume name error in grants table
### Steps to reproduce the behavior (Required)
```
create user jack identified by '123456';
CREATE STORAGE VOLUME
mysv1
TYPE = S3
LOCATIONS = ("s3://xxx/")
COMMENT "default storage"
PROPERTIES(
    "aws.s3.region" = "xxx",
    "aws.s3.endpoint" = "xxx",
    "aws.s3.use_aws_sdk_default_behavior" = "false",
    "aws.s3.use_instance_profile" = "false",
    "aws.s3.access_key" = "xxx",
    "aws.s3.secret_key" = "xxx"
);
GRANT USAGE on storage volume mysv1 to user jack;
show grants for user jack;
```
### Expected behavior (Required)
```
+--------------+-----------------+---------------------------------------------------------------+
| UserIdentity | Catalog         | Grants                                                                                                          |
+--------------+-----------------+---------------------------------------------------------------+
| 'jack'@'%'   | default_catalog | GRANT USAGE ON STORAGE VOLUME mysv1 TO USER 'jack'@'%'|
+--------------+-----------------+---------------------------------------------------------------+
```
### Real behavior (Required)
```
+--------------+-----------------+-------------------------------------------------------------------------------------------------------------+
| UserIdentity | Catalog         | Grants                                                                                                      |
+--------------+-----------------+-------------------------------------------------------------------------------------------------------------+
| 'jack'@'%'   | default_catalog | GRANT USAGE ON STORAGE VOLUME com.starrocks.privilege.StorageVolumePEntryObject@48465bac TO USER 'jack'@'%' |
+--------------+-----------------+-------------------------------------------------------------------------------------------------------------+
```
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
`branch-3.1-12600b6`

### PR DESCRIPTION
**Pull request by TszKitLo40**
Fixes #26856 
After this PR: 
```
MySQL [(none)]> create user jack identified by '123456';
Query OK, 0 rows affected (0.07 sec)

MySQL [(none)]> GRANT USAGE on storage volume storage_volume4 to user jack;
Query OK, 0 rows affected (0.01 sec)

MySQL [(none)]> show grants for user jack;
+--------------+-----------------+------------------------------------------------------------------+
| UserIdentity | Catalog         | Grants                                                           |
+--------------+-----------------+------------------------------------------------------------------+
| 'jack'@'%'   | default_catalog | GRANT USAGE ON STORAGE VOLUME storage_volume4 TO USER 'jack'@'%' |
+--------------+-----------------+------------------------------------------------------------------+
1 row in set (0.03 sec)
```

## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Review by nshangyiming — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 2 / 6 (33.33%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/privilege/StorageVolumePEntryObject.java|2|6|33.33%|[124, 125, 126, 128]|

File: fe/fe-core/src/main/java/com/starrocks/privilege/StorageVolumePEntryObject.java [hunk 1]
```diff
@@ -114,4 +115,17 @@ public int hashCode() {
     public PEntryObject clone() {
         return new StorageVolumePEntryObject(id);
     }
+
+    @Override
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/privilege/StorageVolumePEntryObject.java:119]: better add ut or sql-tester to cover the scenario.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 2 / 6 (33.33%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/privilege/StorageVolumePEntryObject.java|2|6|33.33%|[124, 125, 126, 128]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 2 / 6 (33.33%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/privilege/StorageVolumePEntryObject.java|2|6|33.33%|[124, 125, 126, 128]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 2 / 6 (33.33%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/privilege/StorageVolumePEntryObject.java|2|6|33.33%|[124, 125, 126, 128]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 2 / 6 (33.33%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/privilege/StorageVolumePEntryObject.java|2|6|33.33%|[124, 125, 126, 128]|

File: fe/fe-core/src/main/java/com/starrocks/privilege/StorageVolumePEntryObject.java (hunk 1 above)
Comment (TszKitLo40) [reply fe/fe-core/src/main/java/com/starrocks/privilege/StorageVolumePEntryObject.java:119]: > better add ut or sql-tester to cover the scenario.

done

**Review by HangyuanLiu — APPROVED** [review]

**Review by nshangyiming — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Pull request merged**